docs/self-managed/platform-deployment/openshift-helm.md

```diff
@@ -0,0 +1,194 @@
+
+```bash
+#!/bin/bash -eu
+# Expected usage is as an Helm post renderer.
+# Example usage:
+#   $ helm install my-release camunda/camunda-platform --post-renderer ./patch.sh
+#
+# This script is a Helm chart post-renderer for users on Helm 3.2.0 and greater. It allows removing default
+# values set in sub-charts/dependencies, something which should be possible but is currently not working.
+# See this issue for more: https://github.com/helm/helm/issues/9136
+#
+# The result of patching the rendered Helm templates is printed out to STDOUT. Any other logging from the
+# script is thus sent to STDERR.
+#
+# Note to contributors: this post-renderer is used in the integration tests, so make sure that it can be used
+# from any working directory.
+
+set -o pipefail
+
+# Perform two passes: once for single quotes, once for double quotes, as it's not specified that string values are
+# always output with single or double quotes
+sed -e "s/'@@null@@'/null/g" -e 's/"@@null@@"/null/g'
+```
+
+We will also need to use a custom values file where instead of using `null` as a value to unset default values, we will use a special marker value which will be removed by the post-renderer.
+
+Copy these values to your values file, or better, save them as a separate file, e.g. `openshift.yaml`, also available [here](https://github.com/camunda/camunda-platform-helm/blob/main/openshift/values-patch.yaml):
+
+```yaml
+# omit this section if elasticsearch.enabled is false
+elasticsearch:
+  securityContext:
+    runAsUser: "@@null@@"
+  sysctlInitContainer:
+    enabled: false
+  podSecurityContext:
+    fsGroup: "@@null@@"
+    runAsUser: "@@null@@"
+
+# omit this section if identity.enabled is false
+identity:
+  # omit this section if identity.keycloak.enabled is false
+  keycloak:
+    containerSecurityContext:
+      runAsUser: "@@null@@"
+    podSecurityContext:
+      fsGroup: "@@null@@"
+      runAsUser: "@@null@@"
+    postgresql:
+      # omit this section if identity.keycloak.postgresql.primary.enabled is false
+      primary:
+        containerSecurityContext:
+          runAsUser: "@@null@@"
+        podSecurityContext:
+          fsGroup: "@@null@@"
+          runAsUser: "@@null@@"
+      # omit this section if identity.keycloak.postgresql.readReplicas.enabled is false
+      readReplicas:
+        containerSecurityContext:
+          runAsUser: "@@null@@"
+        podSecurityContext:
+          fsGroup: "@@null@@"
+          runAsUser: "@@null@@"
+      # omit this section if identity.keycloak.postgresql.metrics.enabled is false
+      metrics:
+        containerSecurityContext:
+          runAsUser: "@@null@@"
+        podSecurityContext:
+          fsGroup: "@@null@@"
+          runAsUser: "@@null@@"
+```
+
+Now, when installing the charts, you can do so running:
+
+```shell
+helm install <RELEASE NAME> camunda/camunda-platform --skip-crds -f openshift.yaml -f values.yaml --post-renderer ./patch.sh
+```
+
+:::note
```

Comment:
I think we need `warning` here.

Reply:
:+1: I will also move it a bit higher in the section. I think as the last note it's easy to miss.